cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/requestrate/RequestRateConfiguration.java

```diff
@@ -47,10 +67,48 @@ public static RequestRateConfiguration create(Map<String, Object> options) throw
         RequestRateConfiguration conf = new RequestRateConfiguration();
         Yaml yaml = new Yaml();
         String str = yaml.dumpAsMap(options);
-        conf.values = yaml.loadAs(str, RequestRateConfiguration.Values.class);
+
+        try {
+            conf.values = yaml.loadAs(str, RequestRateConfiguration.Values.class);
+        } catch (Exception e) {
+            throw new ConfigurationException("Unable to load configuration.", e);
+        }
+
+        validateRequestsToReport(conf);
+
         return conf;
     }
 
+    private static void validateRequestsToReport(RequestRateConfiguration conf) throws ConfigurationException {
+        for (String requestToReport : conf.requestsToReport()) {
+            if (ALL_REQUESTS_TO_REPORT.equals(requestToReport)) {
+                continue;
+            }
+
+            String[] requestMeta = requestToReport.split(REQUEST_META_DELIMITER);
+
+            if (requestMeta.length != 2) {
+                throw new ConfigurationException(
+                        "Only two configuration parameters supported, statement type and consistency level.");
+            }
+
+            String statementType = requestMeta[0];
```

Comment:
If you build a map of CL:ST and use pattern matcher you don't need to unwrap the value when matching it to the request rate dictionary key. Should simplify the implementation a lot.